webofneeds/won-owner-webapp/src/main/webapp/app/reducers/need-reducer.js

```diff
@@ -140,6 +140,15 @@ export default function(allNeedsInState = initialState, action = {}) {
         case actionTypes.messages.open.failure:
             return changeConnectionState(allNeedsInState,  action.payload.events['msg:FromSystem'].hasReceiver, won.WON.RequestReceived);
 
+        case actionTypes.messages.open.successRemote:
+        case actionTypes.messages.connect.successRemote:
+            // use the remote success message to obtain the remote connection uri (which we may not have known)
+            var wonMessage = action.payload;
+            var connectionUri =  wonMessage.getReceiver();
+            var needUri =  wonMessage.getReceiverNeed();
+            var remoteConnectionUri = wonMessage.getSender();
+            return allNeedsInState.setIn([needUri, 'connections', connectionUri, 'remoteConnectionUri'], remoteConnectionUri);
```

Comment:
i have some issues with this solution:
-if we add/set the remoteConnectionUri like this we probably produce entries in the connections that only have remoteConnectionUri as the data -> which in return breaks the views that rely on having the correct connection data -> please make a failsafe check to see if the connection is already in the state and then set the remoteConnectionUri accordingly
- Please also update the addConnectionFull method so that it represents the remoteConnectionUri as soon as the connection is added to the state (if the remoteConnectionUri is not present at that given time initialize it with undefined and set it later e.g. here)